jumpscale/entry_points/threebot.py

```diff
@@ -0,0 +1,71 @@
+import click
+
+from jumpscale.loader import j
+from jumpscale.threesdk.identitymanager import IdentityManager
+
+NETWORKS = {"mainnet": "explorer.grid.tf", "testnet": "explorer.testnet.grid.tf", "devnet": "explorer.devnet.grid.tf"}
+
+
+@click.command()
+@click.option("--identity", default=None, help="threebot name(i,e name.3bot)")
+def start(identity=None):
+    """start 3bot server after making sure identity is ok
+    It will start with the default identity in j.me, if you'd like to specify an identity
+    please pass the optional arguments
+
+    usage: threebot start
+    or: threebot start --identity <identity> --explorer <explorer>
+
+    Args:
+        identity (str, optional): threebot name. Defaults to None.
+        explorer (str, optional): which explorer network to use: mainnet, testnet, devnet. Defaults to None.
+    """
+    if identity:
+        if j.core.identity.find(identity):
+            print(f"Setting default identity to {identity}\nStarting threebot server...")
+            j.core.identity.set_default(identity)
+            j.core.identity.me.save()
+        else:
+            j.tools.console.printcolors(
+                "{RED}Identity %s is not set, please configure it or start with the default one" % identity
+            )
+            return
+
+    if not j.core.identity.list_all():
+        identity_data = IdentityManager()
+        identity, email, words, explorer = identity_data.ask_identity()
+        me = j.core.identity.new(
+            "default", tname=identity, email=email, words=words, explorer_url=f"https://{explorer}/explorer"
+        )
+        me.register()
+        me.save()
+
+    cmd = j.tools.startupcmd.get("threebot_default")
+    cmd.start_cmd = "jsng 'j.servers.threebot.start_default(wait=True)'"
+    cmd.process_strings_regex = [j.tools.nginx.get("default").check_command_string, "nginx.*"]
+    cmd.ports = [8000, 8999]
+    cmd.start()
+    print("\n✅ Threebot server started\n")
+    j.tools.console.printcolors("{WHITE}Visit admin dashboard at: {GREEN}http://localhost/\n")
```

Comment:
if we print this like that how would this look if invoked from 3sdk, where they actually have to access the admin portal on another port :-S? @xmonader suggestions?

Reply:
problem is it was consistent in jsx because it was always 4000 and used via port forward so maybe we can do port check? as a simple check or check for startup_cmd running on the host namespace? if that the case we can show the localhost message?

Reply:
will only print in host